src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -205,7 +205,72 @@ def _get_resource_group(self, **kwargs):
         except KeyError:
             template = 'To create a KeyVault a resource group is required. Please add ' \
                        'decorator @{} in front of this KeyVault preparer.'
-            raise CliTestError(template.format(KeyVaultPreparer.__name__))
+            raise CliTestError(template.format(ResourceGroupPreparer.__name__))
+
+
+# Managed HSM Preparer and its shorthand decorator
+
+# pylint: disable=too-many-instance-attributes
+class ManagedHSMPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
+    def __init__(self, certs_path, name_prefix='clitest', location='uksouth', key='hsm', name_len=24,
+                 parameter_name='managed_hsm', resource_group_parameter_name='resource_group',
+                 administrators=None, additional_params=None):
+        super(ManagedHSMPreparer, self).__init__(name_prefix, name_len)
+        self.cli_ctx = get_dummy_cli()
+        self.location = location
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.parameter_name = parameter_name
+        self.key = key
+        self.certs_path = certs_path
+        self.administrators = administrators
+        self.additional_params = additional_params
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        administrators = self.administrators or self._get_signed_in_user()
+        template = 'az keyvault create --hsm-name {} -g {} -l {} --administrators {} --retention-days 7'
+        if self.additional_params:
+            template += self.additional_params
+        self.live_only_execute(self.cli_ctx, template.format(name, group, self.location, administrators))
+        # After creating MHSM, All data plane commands are disabled until the HSM is activated.
+        # To activate the HSM, we must download the Security Domain.
+        if self.certs_path:
+            cert0 = os.path.join(self.certs_path, 'cert_0.cer').replace('\\', '\\\\')
+            cert1 = os.path.join(self.certs_path, 'cert_1.cer').replace('\\', '\\\\')
+            cert2 = os.path.join(self.certs_path, 'cert_2.cer').replace('\\', '\\\\')
```

Comment:
Let's add the `openssl`s command that are used to generate the cert for future reference.

Reply:
Added the MHSM activation reference doc which contains how to generate self-signed certificates.